mne/annotations.py

```diff
@@ -567,6 +567,49 @@ def crop(self, tmin=None, tmax=None, emit_warning=False, verbose=None):
 
         return self
 
+    @verbose
+    def set_durations(self, mapping, verbose=None):
+        """Set annotation durations(s). Operates inplace.
+
+        Parameters
+        ----------
+        mapping : dict | float | int
+            A dictionary mapping the annotation description to a duration
+            e.g. ``{'ShortStimulus' : 3, 'LongStimulus' : 12}``.
+            Alternatively, if a number is provided, then all annotations
+            durations are set to the single provided value.
+        %(verbose_meth)s
+
+        Returns
+        -------
+        self : mne.Annotations
+            The modified Annotations object.
+
+        Notes
+        -----
+        .. versionadded:: 0.24.0
+        """
+        if isinstance(mapping, dict):
+            orig_annots = sorted(list(mapping.keys()))
+            missing = [key not in self.description for key in orig_annots]
+            if any(missing):
+                raise ValueError(
+                    "Annotation description(s) in mapping missing from data: "
+                    "%s" % np.array(orig_annots)[np.array(missing)])
+            for map in mapping:
+                map_idx = [desc == map for desc in self.description]
+                self.duration[map_idx] = mapping[map]
+
+        elif isinstance(mapping, (int, float)):
```

Comment:
this is dangerous as it will fail eg for an np.int32. You should use:

isinstance(mapping, numbers.Number)

Reply:
For consistency you should use `_is_numeric(mapping)` from `.utils`. Maybe we should just use `numbers.Number` there, too, but we can do that later